indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -469,9 +469,15 @@ public Response getMultipleTaskStatuses(Set<String> taskIds)
       return Response.status(Response.Status.BAD_REQUEST).entity("No TaskIds provided.").build();
     }
 
+    final Optional<TaskQueue> taskQueue = taskMaster.getTaskQueue();
     Map<String, TaskStatus> result = Maps.newHashMapWithExpectedSize(taskIds.size());
     for (String taskId : taskIds) {
-      Optional<TaskStatus> optional = taskStorageQueryAdapter.getStatus(taskId);
+      final Optional<TaskStatus> optional;
+      if (taskQueue.isPresent()) {
```

Comment:
If the task is not present in the `taskQueue`, we should then check the `taskStorageQueryAdapter`. It might be already complete.

Reply:
TaskQueue#getStatus looks at the metadata store if the task is not running

```
  public Optional<TaskStatus> getTaskStatus(final String taskId)
  {
    RunnerTaskState runnerTaskState = taskRunner.getRunnerTaskState(taskId);
    if (runnerTaskState != null && runnerTaskState != RunnerTaskState.NONE) {
      return Optional.of(TaskStatus.running(taskId).withLocation(taskRunner.getTaskLocation(taskId)));
    } else {
      return taskStorage.getStatus(taskId);
    }
  }
```

Would it be better to move this fallback to db logic out of the TaskQueue in the calling method?

Reply:
Ah, I see, I didn't realize that the fallback logic was in the `TaskQueue` method. In that case this code is OK.